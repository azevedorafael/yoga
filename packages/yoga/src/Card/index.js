import Card from './web/Card';
import CardFooter from './web/CardFooter';
import CardHeader from './web/CardHeader';
import Plan from './web/Plan';

Card.Footer = CardFooter;
Card.Header = CardHeader;
Card.Plan = Plan;

export default Card;
