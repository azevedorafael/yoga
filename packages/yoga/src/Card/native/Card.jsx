import React from 'react';
import styled from 'styled-components';

const CardStyled = styled.View(
  ({
    theme: {
      yoga: {
        components: { card },
      },
    },
  }) => `
    padding:
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${card.backgroundColor};
    box-shadow: ${card.elevation};
    elevation: 4;
`,
);

const Card = ({ ...rest }) => <CardStyled {...rest} />;

Card.displayName = 'Card';

export default Card;
