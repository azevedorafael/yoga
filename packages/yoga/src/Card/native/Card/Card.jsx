import React from 'react';
import styled from 'styled-components';
import { string, shape, oneOf, node } from 'prop-types';

import Text from '../../../Text';

const CardStyled = styled.View(
  ({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        colors: { [variant]: color = {} },
        components: { card },
      },
    },
  }) => `
    padding:
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${
      variant ? color[variantIntensity] : card.backgroundColor
    };
    box-shadow: ${card.elevation};
    elevation: 4;
`,
);

const Ribbon = styled.View`
  ${({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        components: { card },
        spacing,
        colors: { gray, [variant]: color = {}, white, dark },
      },
    },
  }) => `
    align-self: flex-start;

    margin-bottom: ${card.padding.bottom}px;
    margin-left: -${card.padding.left}px;

    padding: 
      ${spacing.xxsmall}px
      ${spacing.small}px
      ${spacing.xxsmall}px
      ${spacing.medium}px;
    
    border-top-right-radius: ${card.ribbon.radius}px;
    border-bottom-right-radius: ${card.ribbon.radius}px;
    background-color: ${
      variant
        ? color[typeof variantIntensity === 'number' ? variantIntensity : 3]
        : gray[1]
    };

    color: ${variant ? white : dark};
  `}
`;

const RibbonText = styled(Text.Tiny)`
  ${({
    variant,
    theme: {
      yoga: {
        colors: { white, dark },
      },
    },
  }) => `
    color: ${variant ? white : dark};
  `}
`;

const Card = ({ ribbon, children, ...rest }) => (
  <CardStyled {...rest}>
    {Object.keys(ribbon).length > 0 && (
      <Ribbon
        variant={ribbon.variant}
        variantIntensity={ribbon.variantIntensity}
      >
        <RibbonText variant={ribbon.variant}>{ribbon.text}</RibbonText>
      </Ribbon>
    )}
    {children}
  </CardStyled>
);

Card.propTypes = {
  /** text: the content inside the Card Ribbon
   * variant: style the ribbon following the theme (primary, secondary, tertiary)
   * variantIntensity: ribbon variant color intensity (0, 1, 2, 3) */
  ribbon: shape({
    text: string,
    variant: oneOf(['', 'primary', 'secondary']),
    variantIntensity: oneOf([0, 1, 2, 3]),
  }),
  children: node,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['', 'primary', 'secondary']),
  /** intensity of variant color (0, 1, 2, 3) */
  variantIntensity: oneOf([0, 1, 2, 3]),
};

Card.defaultProps = {
  ribbon: {},
  children: null,
  variant: '',
  variantIntensity: 3,
};

Card.displayName = 'Card';

export default Card;
