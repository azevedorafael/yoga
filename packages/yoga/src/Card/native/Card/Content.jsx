import styled from 'styled-components';

const Content = styled.View``;

Content.displayName = 'Card.Content';

export default Content;
