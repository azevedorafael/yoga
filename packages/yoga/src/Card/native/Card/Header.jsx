import styled from 'styled-components';

const Header = styled.View``;

Header.displayName = 'Card.Header';

export default Header;
