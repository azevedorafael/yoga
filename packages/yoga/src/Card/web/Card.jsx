import React from 'react';
import styled from 'styled-components';

const CardStyled = styled.div(
  ({
    theme: {
      components: { card },
    },
  }) => `
    display: flex;
    flex-direction: column;
    justify-content: flex-start;

    width: 100%;
    padding: 
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${card.background};
    box-shadow: ${card.elevation};
`,
);

const Card = ({ ...rest }) => <CardStyled {...rest} />;

Card.displayName = 'Card';

export default Card;
