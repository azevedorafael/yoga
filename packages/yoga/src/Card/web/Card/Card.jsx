import React from 'react';
import styled from 'styled-components';
import { string, shape, oneOf, node } from 'prop-types';

import Text from '../../../Text';

const CardStyled = styled.section`
  ${({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        colors: { [variant]: color = {} },
        components: { card },
      },
    },
  }) => `
    padding: 
      ${card.padding.top}px 
      ${card.padding.right}px 
      ${card.padding.bottom}px 
      ${card.padding.left}px;

    border-radius: ${card.radii}px;
    background-color: ${
      variant ? color[variantIntensity] : card.backgroundColor
    };
    box-shadow: ${card.elevation};
  `}
`;

const Ribbon = styled(Text.Tiny)`
  ${({
    variant,
    variantIntensity,
    theme: {
      yoga: {
        components: { card },
        spacing,
        colors: { gray, [variant]: color = {}, white },
      },
    },
  }) => `
    display: inline-block;
    top: ${card.padding.top}px;
    left: 0;

    margin-left: -${card.padding.left}px;
    margin-bottom: ${card.padding.bottom}px;

    padding: 
      ${spacing.xxsmall}px
      ${spacing.small}px
      ${spacing.xxsmall}px
      ${spacing.medium}px;
    
    border-top-right-radius: ${card.ribbon.radius}px;
    border-bottom-right-radius: ${card.ribbon.radius}px;

    background-color: ${
      variant
        ? color[typeof variantIntensity === 'number' ? variantIntensity : 2]
        : gray.i10
    };

    color: ${variant ? white : gray.i80};
  `}
`;

const Card = ({ ribbon, children, ...rest }) => (
  <CardStyled {...rest}>
    {Object.keys(ribbon).length > 0 && (
      <Ribbon
        variant={ribbon.variant}
        variantIntensity={ribbon.variantIntensity}
        as="span"
      >
        {ribbon.text}
      </Ribbon>
    )}
    {children}
  </CardStyled>
);

Card.propTypes = {
  /** text: the content inside the Card Ribbon
   * variant: style the ribbon following the theme (primary, secondary, tertiary)
   * variantIntensity: ribbon variant color intensity (0, 1, 2) */
  ribbon: shape({
    text: string,
    variant: oneOf(['', 'primary', 'secondary']),
    variantIntensity: oneOf([0, 1, 2]),
  }),
  children: node,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['', 'primary', 'secondary']),
  /** intensity of variant color (0, 1, 2) */
  variantIntensity: oneOf([0, 1, 2]),
};

Card.defaultProps = {
  ribbon: {},
  children: null,
  variant: '',
  variantIntensity: 2,
};

Card.displayName = 'Card';

export default Card;
