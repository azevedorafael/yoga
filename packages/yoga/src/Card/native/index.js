import Card from './Card';
import PlanCard from './PlanCard';
import EventCard from './EventCard';
import GymCard from './GymCard';

export { Card, PlanCard, EventCard, GymCard };
