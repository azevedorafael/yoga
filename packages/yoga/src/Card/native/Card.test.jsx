import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import ThemeProvider from '../../ThemeProvider';
import Card from '..';

const data = {
  plan: {
    name: 'Basic',
    price: 'U$ 19.90',
    period: 'month',
    gyms: '1,239',
  },
  ribbonDefault: { label: 'Upgrade' },
  ribbonSecondary: { label: 'Upgrade', secondary: true },
};

describe('<Card />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default Card', () => {
      const { container } = render(
        <ThemeProvider>
          <Card>
            <Text>Hello World</Text>
          </Card>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe.skip('Card Plan Tests', () => {
    it('should match texts with Plan Card', () => {
      console.log(CardPlan);
      const { getByText } = render(
        <ThemeProvider>
          <Card.Plan plan={data.plan} />
        </ThemeProvider>,
      );
      expect(getByText(data.plan.name)).toBeTruthy();
      expect(getByText(data.plan.price)).toBeTruthy();
      expect(getByText(`/${data.plan.period}`)).toBeTruthy();
      expect(getByText(`${data.plan.gyms} gyms`)).toBeTruthy();
    });

    it('should match text Plan Card with ribbon default', () => {
      const { getByText } = render(
        <ThemeProvider>
          <Card.Plan plan={data.plan} ribbon={data.ribbonDefault} />
        </ThemeProvider>,
      );
      expect(getByText(data.ribbonDefault.label)).toBeTruthy();
    });

    it('should match text Plan Card with ribbon secondary', () => {
      const { getByText } = render(
        <ThemeProvider>
          <Card.Plan plan={data.plan} ribbon={data.ribbonSecondary} />
        </ThemeProvider>,
      );
      expect(getByText(data.ribbonSecondary.label)).toBeTruthy();
    });

    it('should match snapshot Plan Card with selected boolean prop', () => {
      const { container } = render(
        <ThemeProvider>
          <Card.Plan selected plan={data.plan} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should call onCardPress function when press on Card.Plan', () => {
      const onCardPressMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Card.Plan
            plan={data.plan}
            onPress={onCardPressMock}
            testID="touchable"
          />
        </ThemeProvider>,
      );
      fireEvent.press(getByTestId('touchable'));
      expect(onCardPressMock).toHaveBeenCalled();
    });
  });
});
