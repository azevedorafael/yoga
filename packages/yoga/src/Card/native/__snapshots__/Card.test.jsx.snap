// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot Plan Card with selected boolean prop 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    selected={true}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "display": "flex",
          "elevation": 4,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#178173",
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      testID="touchable"
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "position": "relative",
              "width": "100%",
            },
            Object {
              "marginBottom": 40,
            },
          ]
        }
      >
        <View
          ribbon={null}
          style={
            Array [
              Object {
                "display": "flex",
                "flexBasis": "100%",
                "flexDirection": "row",
                "flexWrap": "wrap",
              },
            ]
          }
        >
          <Text
            selected={true}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 12,
                },
              ]
            }
          >
            Basic
          </Text>
        </View>
      </View>
      <Text
        selected={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "fontWeight": "700",
            },
          ]
        }
      >
        U$ 19.90
      </Text>
      <Text
        selected={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
              "paddingTop": 4,
            },
          ]
        }
      >
        /
        month
      </Text>
      <View
        style={
          Array [
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            },
            Object {
              "marginTop": 20,
            },
          ]
        }
      >
        <Text
          selected={true}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "700",
              },
            ]
          }
        >
          1,239
           gyms
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`<Card /> should match snapshot with default Card 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "display": "flex",
          "elevation": 4,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
          "width": "100%",
        },
      ]
    }
  >
    <Text>
      Teste
    </Text>
  </View>
</View>
`;
